File: src/main/java/dev/esophose/discordbot/command/arguments/ColorArgumentHandler.java
package dev.esophose.discordbot.command.arguments;

import dev.esophose.discordbot.command.DiscordCommandArgumentHandler;
import discord4j.core.object.entity.Guild;
import java.awt.Color;
import reactor.core.publisher.Mono;

public class ColorArgumentHandler extends DiscordCommandArgumentHandler<Color> {

    @Override
    protected Mono<Color> handleInternal(Guild guild, String input) {
        try {
            return Mono.just(Color.decode(input));
        } catch (Exception ex) {
            return Mono.empty();
        }
    }

    @Override
    public String getErrorMessage(Guild guild, String input) {
        return "Invalid color: [" + input + "], make sure it starts with # if using a hex code";
    }

    @Override
    public Class<Color> getHandledType() {
        return Color.class;
    }

}

File: src/main/kotlin/dev/esophose/discordbot/command/arguments/ColorArgumentHandler.kt
package dev.esophose.discordbot.command.arguments

import dev.esophose.discordbot.command.DiscordCommandArgumentHandler
import discord4j.core.`object`.entity.Guild
import discord4j.rest.util.Color
import reactor.core.publisher.Mono
import kotlin.reflect.KClass

class ColorArgumentHandler : DiscordCommandArgumentHandler<Color>() {

    override val handledType: KClass<Color>
        get() = Color::class

    override fun handleInternal(guild: Guild, input: String): Mono<Color> {
        return try {
            Mono.just(Color(java.awt.Color.decode(input).rgb))
        } catch (ex: Exception) {
            Mono.empty()
        }
    }

    override fun getErrorMessage(guild: Guild, input: String): String {
        return "Invalid color: [$input], make sure it starts with # if using a hex code"
    }

}
